src/Interpreters/Context.cpp

```diff
@@ -3417,9 +3417,8 @@ void Context::initializeSystemLogs()
     /// triggered from another thread, that is launched while initializing the system logs,
     /// for example, system.filesystem_cache_log will be triggered by parts loading
     /// of any other table if it is stored on a disk with cache.
-    callOnce(shared->system_logs_initializer, [&] {
-        shared->system_logs = std::make_unique<SystemLogs>(getGlobalContext(), getConfigRef());
-    });
+    auto lock = getGlobalLock();
+    shared->system_logs = std::make_unique<SystemLogs>(getGlobalContext(), getConfigRef());
```

Comment:
If this fix pass tests we need to remove `shared->system_logs_initializer`.
But I suspect this can lead to deadlock because SystemLogs constructor can call context methods.
Proper fix should be something like this:
```
callOnce(shared->system_logs_initializer, [&] {
    auto system_logs = std::make_unique<SystemLogs>(getGlobalContext(), getConfigRef());
    auto lock = getGlobalSharedLock();
    shared->system_logs = std::move(system_logs);
});
```